### ISSUE TITLE
Webpack and Babel are broken.

### ISSUE THREAD

**Issue by kiranoot** [zwagoth/e621ng#314]
Because rails/node/node package changes have forced an unintended upgrade of node/webpack/webpacker in order to function at all(thanks random package that dropped older nodejs versions in a point release, and thanks packages that don't compile with newer nodejs versions)

As of right now there are many known breakages that need looking into:
 - Usage of JS Sets is broken because they are internally converted into a string instead of remaining a set. This may be incorrect JS usage, or it may be babel compiling code related to sets incorrectly. Most likely this is some tuning to babel that needs modifications.
 - By default JS is attempted to be loaded in deferred mode, this broke all browser extensions, and some of the on site code that expect the Danbooru object to exist on startup, and so has been disabled.
 - Babel targets and other transpiler options are all suspect at this time and need to be targeted back to ES5 if possible to match existing promises to the community.

Because of the size of the codebase and the inability to roll back this change once we commit to it(again, it was forced upon us unexpected in a point release of a package that is required for the site to run.) any help locating broken pieces of javascript and help resolving them is appreciated.

**Comment by kiranoot** [comment]
Investigation suggests that this is most likely related to the existing babel config, plugins and usage of "loose" which produces incorrect code when applied against Set and the spread operator. Targeting a different target, such as Chrome 40 may also be beneficial in order to ensure that code doesn't use newer features when it doesn't need to.

**Comment by kiranoot** [comment]
This should be fixed with cb84bf4c9b0439f12e3b3a7ce9486bbf10dfd923 however, a second set of eyes and some additional testing would be helpful, as it's a major change to how JS and CSS are produced.

Under this new setup minification does not take place unless rails is in the production config, even when precompiling assets. Make sure to run precompile under the correct rails env if testing minification is behaving properly.

**Comment by Earlopain** [comment]
The config is mostly giberish for me, I'm not sure whats going on there but I'll take some time to test things through.
I'll do that sometime today or tomorrow and write if I found any breakage or not.

**Comment by Earlopain** [comment]
I took a good amount of time to test all aspects of the site but everything else seems to work as intended.

- Blacklist
- Voting/Faving
- Uploading
- Post moderation (flagging, deletion, disapprovals etc.)
- Shortcuts
- Notes
- Autocomplete

Those are the things I tested in more detail but generally I looked at which js files there are and where they might be used and triggered their function. Things do seem to be working with this change.
